Products/ZenRRD/zencommand.py

```diff
@@ -947,21 +932,25 @@ def makeCmdTimeoutEvent(deviceId, datasource, severity=None):
         deviceId,
         datasource,
         "Datasource %s command timed out" % datasource.name,
-        severity=severity,
+        severity=Error if severity is None else severity,
         eventKey="CommandTimeout",
+        component=datasource.name,
```

Comment:
worried about re-using component here to represent the datasource, as it is possible for a device-level datasource to have the same name as an actual device component.  Could pass "" instead to have this be a device-level event for a component template.

Reply:
Having the component field be empty is the reason why timeout events are getting cleared.  Two different datasources will fire events that look identical.  The event identity is based on the device, event class, component, and event key.  For all the datasources, the device, event class, and event key are identical.  The only way to differentiate is to use the datasource for the component.

Reply:
Since you are setting a specific event key for timeouts, if you need to differentiate one datasource's timeout from another's, just prepend the timeout event key with the datasource?

Reply:
I can do that.

Looking at other examples of event key values, I see:

interval_exceeded
snmp_v3_error
snmp_error
partial_oids_collected
no_oid_results
agent_down
cycle_interval
correlation_execution
nmap_execution
ping_lag
transform_failed
transform_disabled
zCommandUsername

so why not \<datasource>\_timeout (or timeout\_\<datasource>).